text/0000-embedded-refract.md

```diff
@@ -0,0 +1,361 @@
+  "refract": {
+    "element": "object",
+    "meta": {
+      "id": "bowler-103"
+    },
+    "content": {
+      "age": {
+        "element": "number",
+        "meta": {
+          "id": "bowner-103-age"
+        },
+        "content": 28
+      }
+    }
+  },
+  "first_name": "John",
+  "last_name": "Doe",
+  "scores": [150, 202, 145]
+}
+```
+
+This will also support full member elements, so even this below is equivalent (note that it's a slight modification from the full Refract):
+
+```json
+{
+  "refract": {
+    "element": "object",
+    "meta": {
+      "id": "bowler-103"
+    },
+    "content": [
+      {
+        "element": "member",
+        "content": {
+          "key": "age",
+          "value": {
+            "refract": {
+              "element": "number",
+              "meta": {
+                "id": "bowner-103-age"
+              },
+              "content": 28
+            }
+          }
+        }
+      }
+    ]
+  },
+  "first_name": "John",
+  "last_name": "Doe",
+  "scores": [150, 202, 145]
+}
+```
+
+In full Refract, you would have to also provide the element for the key "age" in the previous example, but since we can rely on the `refract` keyword, we can leave it unrefracted. We could Refract it more later if necessary.
+
+## Refracting Array and Array Items
+
+Array items can be embedded without refracting the entire array.
+
+```json
+{
+  "first_name": "John",
+  "last_name": "Doe",
+  "age": 28,
+  "scores": [
+    {
+      "refract": {
+        "element": "number",
+        "meta": {
+          "id": "bowler-103-game-1"
+        },
+        "content": 150
+      }
+    },
+    202,
+    145]
+}
+```
+
+Full arrays can also be refracted.
+
+```json
+{
+  "first_name": "John",
+  "last_name": "Doe",
+  "age": 28,
+  "scores": {
+    "refract": {
+      "element": "array",
+      "meta": {
+        "id": "bowler-103-scores"
+      },
+      "content": [150, 202, 145]
+    }
+  }
+}
+```
+
+## Refracting Attributes
+
+This makes it nice for refracting attributes, because now you know what attributes have been refracted. This could become:
+
+```json
+{
+  "refract": {
+    "element": "object",
+    "meta": {
+      "id": "bowler-103"
+    },
+    "attributes": {
+      "foo": "baz"
+    }
+  },
+  "first_name": "John",
+  "last_name": "Doe",
+  "age": 28,
+  "scores": [150, 202, 145]
+}
+```
+
+```json
+{
+  "refract": {
+    "element": "object",
+    "meta": {
+      "id": "bowler-103"
+    },
+    "attributes": {
+      "foo": {
+        "refract": {
+          "element": "string",
+          "content": "baz"
+        }
+      }
+    }
+  },
+  "first_name": "John",
+  "last_name": "Doe",
+  "age": 28,
+  "scores": [150, 202, 145]
+}
+```
+
+## Edge cases
+
+### Properties in Value Elements
+
+What would this mean, where we have the `foo` property?
+
+```json
+{
+  "first_name": "John",
+  "last_name": "Doe",
+  "age": {
+    "refract": {
+      "element": "number",
+      "content": 28
+    },
+    "foo": "bar"
```

Comment:
We may want to disallow this if it seems confusing enough and just treat it as an error. There's already a lot of magic going on :wink: